packages/pyright-internal/src/analyzer/typeEvaluator.ts

```diff
@@ -15409,11 +15409,17 @@ export function createTypeEvaluator(
         }
 
         // Or if the object is in an untyped library that was explicitly mentioned.
-        if (type.shared && 'moduleName' in type.shared) {
-            const moduleName = type.shared.moduleName;
-            if (moduleIsInList(ruleset.allowedUntypedLibraries, moduleName)) {
-                return;
-            }
+        // Overloaded methods must be handled separately, as the module is stored on the implementation.
```

Comment:
we should probably explain in this comment why we aren't getting the module from the implementation
```suggestion
        // Overloaded methods must be handled separately, as the module is stored on the implementation and the overloads.
        // we get the module from the first overload in case there's no implementation (eg. if the overload came from a stub file)
```